warehouse/forklift/legacy.py

```diff
@@ -1274,8 +1316,15 @@ def file_upload(request):
                 "The digest supplied does not match a digest calculated "
                 "from the uploaded file.",
             )
-        # Skip duplicate check for files when it's a draft release
-        if not release.is_draft:
+        # Skip duplicate check for files when it's a draft release,
+        # and delete existing files instead
+        if release.is_draft:
+            existing_file = (
+                request.db.query(File).filter(File.filename == filename).first()
+            )
+            if existing_file is not None:
+                request.db.delete(existing_file)
```

Comment:
This kinda answers my previous question on whether both files were kept. I could be wrong, but I believe deleting the File doesn't delete it from storage, which could be a problem because we're now responsible for a file that we don't know about. If we get, e.g. a legal order to remove all releases from a project because of a copyright infringement, we won't be able to make sure that we did so.

(not 100% sure that deleting the file doesn't delete it from storage too, but I wasn't able to be 100% sure of that, and it looks like it doesn't)

Reply:
Well, if it's done this way elsewhere, I'm sold.

(I'd still be interested in how we manage copyright stuff, but at least we're consistent)